lxd/device/disk.go

```diff
@@ -1348,7 +1362,7 @@ func (d *disk) startVM() (*deviceConfig.RunConfig, error) {
 					}
 				}
 			} else {
```

Comment:
should we have a check for isPath here, and then an else that returns an error about unexpected type if no matching type at that stage?